### PR TITLE
Add clang 3.8.1

### PR DESCRIPTION
**Pull request by alarcher**
---

Failing Tests (5):
    Clang :: Driver/cl-response-file.c
    Clang :: Sema/wchar.c
    LLVM :: CodeGen/AMDGPU/fminnum.ll
    LLVM-Unit :: ADT/Release+Asserts/ADTTests/APIntTest.LargeAPIntConstruction
    LLVM-Unit :: ADT/Release+Asserts/ADTTests/APIntTest.nearestLogBase2

  Expected Passes    : 24174
  Expected Failures  : 163
  Unsupported Tests  : 161
  Unexpected Failures: 5

### CODE REVIEW

File: components/developer/clang-38/Makefile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2014 Alexander Pyhalov.  All rights reserved.
+#
```
Comment (pyhalov) [inline components/developer/clang-38/Makefile:14]: Fix copyright

**Comment by pyhalov** [comment]
Have you tried actually to compile something with this clang? clang itself?

**Comment by alarcher** [comment]
Ran the test suite and now testing on simple test cases.
Some patches for dealing with paths to be pushed.

File: components/developer/clang-38/Makefile [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+# Copyright 2016 Aurelien Larcher.  All rights reserved.
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME= clang-38
+LLVM_NAME=llvm
+CLANG_NAME=cfe
+LIBCXX_NAME=libcxx
```
Comment (pyhalov) [inline components/developer/clang-38/Makefile]: I'm sorry, it's my error initially, but we don't use libcxx for clang, you can remove this line

Comment (alarcher) [inline components/developer/clang-38/Makefile]: No problem. I have fixed include and library paths in an additional patch and I have tested the C and C++ compiler but I have lost the connection to the build machine right now. I will update the PR when I get access again.

File: components/developer/clang-38/Makefile [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+COMPONENT_VERSION= 3.8.1
+COMPONENT_MAJOR_VERSION= \
+  $(shell echo $(COMPONENT_VERSION) | $(GSED) -e 's/\([0-9]\+\.[0-9]\+\).*/\1/')
+COMPONENT_PROJECT_URL = http://llvm.org/
+COMPONENT_FMRI= developer/clang-3.8
```
Comment (pyhalov) [inline components/developer/clang-38/Makefile:26]: Perhaps, name it developer/clang-38 to be consistent and also rename clang-3.4 to clang-38 or mark obsolete?

**Comment by alarcher** [comment]
Happy testing.

**Comment by alarcher** [comment]
@xen0l Same idea as gcc-5 applies to clang-38.

**Comment by xen0l** [comment]
@pyhalov are you OK with that?

**Comment by pyhalov** [comment]
@alarcher , have you looked at pdns-recursor or openttd?

**Comment by pyhalov** [comment]
sorry , +1 for clang, if @alarcher thinks that it's fine. The comment meant for gcc

**Pull request merged**